File: 1218-longest-arithmetic-subsequence-of-given-difference/1218-longest-arithmetic-subsequence-of-given-difference.java
class Solution {
    public int longestSubsequence(int[] arr, int difference) {
        Map<Integer, Integer> dp = new HashMap<>();
        int ans = 1;
        for(int a : arr){
            int beforeA = dp.getOrDefault(a-difference, 0);
            dp.put(a, beforeA+1);
            ans = Math.max(ans, dp.get(a));
        }
        return ans;
    }
}

File: 1218-longest-arithmetic-subsequence-of-given-difference/1218-longest-arithmetic-subsequence-of-given-difference.py
class Solution:
    def longestSubsequence(self, arr: List[int], difference: int) -> int:
        dp = {}
        ans = 1
        for a in arr:
            beforeA = dp.get(a-difference, 0)
            dp[a] = beforeA + 1
            ans = max(ans, dp[a])
        return ans